### DISCUSSION TITLE
Log stream return empty or eof causes SSE client to terminate the connection

### DISCUSSION THREAD

**Discussion by pi3ch** [woodpecker-ci/woodpecker#2989] [Ask a Question]
I am trying to write a SSE client to get [Log stream](https://woodpecker-ci.org/api#tag/Pipeline-logs/paths/~1stream~1logs~1%7Brepo_id%7D~1%7Bpipeline%7D~1%7BstepID%7D/get), but eventhough [events](https://woodpecker-ci.org/api#tag/Events/paths/~1stream~1events/get) stream shows the step as running connecting to log stream often returns empty or `eof` data. This causes SSE client to terminate the connection. This usually happen when I immediately connect to log stream when step status is ready.

Just checking if anyone else observe this behavior? Any workaround?
Is there any alternative way to get the log streams?

**Comment by anbraten** [comment]
There could be issues with the log-streaming its still not perfectly implemented. AFAIK the log-stream is opened as soon as the first log message is printed as this could take a short time (buffering, etc) you could run into a closed / not yet opened stream.

**Reply by pi3ch** [reply]
It makes challenging to find when exactly to connect so it doesn't terminate. Currently I have a workaround to sleep for a few second after a step set is 'running' before connecting to log stream. I don't recall I had this issue with Drone. Do we know what has changed?
